sqlglot/dialects/duckdb.py

```diff
@@ -141,6 +142,49 @@ def _build_make_timestamp(args: t.List) -> exp.Expression:
     )
 
 
+def _build_greatest(args: t.List) -> exp.Greatest:
+    """Build GREATEST with all arguments properly distributed."""
+    return exp.Greatest(this=seq_get(args, 0), expressions=args[1:])
+
+
+def _approx_quantiles_sql(self: DuckDB.Generator, expression: exp.ApproxQuantiles) -> str:
+    """
+    BigQuery's APPROX_QUANTILES(expr, n) returns an array of n+1 approximate quantile values
+    dividing the input distribution into n equal-sized buckets.
+
+    Both BigQuery and DuckDB use approximate algorithms for quantile estimation, but BigQuery
+    does not document the specific algorithm used so results may differ. DuckDB does not
+    support RESPECT NULLS.
+    """
+    this = expression.this
+    if isinstance(this, exp.Distinct):
+        # APPROX_QUANTILES requires 2 args and DISTINCT node grabs both
+        if len(this.expressions) < 2:
+            self.unsupported("APPROX_QUANTILES requires a bucket count argument")
+            return self.function_fallback_sql(expression)
+        num_quantiles_expr = this.expressions.pop(1)
+    else:
+        num_quantiles_expr = expression.expression
+
+    if isinstance(num_quantiles_expr, (exp.Cast, exp.TryCast)):
+        num_quantiles_expr = num_quantiles_expr.this.unnest()
+
+    if not isinstance(num_quantiles_expr, exp.Literal) or not num_quantiles_expr.is_number:
+        self.unsupported("APPROX_QUANTILES bucket count must be a positive integer")
+        return self.function_fallback_sql(expression)
+
+    num_quantiles = num_quantiles_expr.to_py()
+    if not num_quantiles or not isinstance(num_quantiles, int) or num_quantiles <= 0:
```

Comment:
Should you do `is_int` in L171 instead of checking for an `int` instance here? So the condition then beocmes `if num_quantiles <= 0`.